File: primes.r
library(compiler)
invisible(enableJIT(3))
cat(2)
limit = strtoi(commandArgs(trailingOnly=TRUE)[1]) - 1
half = limit / 2
sieve = array(TRUE, half)
total = 1
if (limit > 7)
	for (i in seq(3, as.integer(sqrt(limit)), 2))
		if (sieve[i / 2]) sieve[seq(i^2 / 2, half, i)] = FALSE
for (i in 1:half)
	if (sieve[i]) {
		total = total + 1
		cat(",", 2 * i + 1)
	}
cat("\nTotal:", total, "\n")

File: primes.js
#!/usr/bin/env node
process.stdout.write("2")
half = process.argv[2] >>> 1
sieve = new Uint8Array(half)
for (i = 3; i < (isqrt = Math.sqrt(process.argv[2])); i += 2)
	if (!sieve[i >>> 1])
		for (j = i ** 2 >>> 1; j < half; j += i) sieve[j] = true
total = 1
for (i = 1; i < half; i++)
	if (!sieve[i]) {
		process.stdout.write(`, ${2 * i + 1}`)
		total++
	}
console.log("\nTotal:", total)
